org/mozilla/jss/ssl/javax/JSSParameters.java

```diff
@@ -190,4 +196,27 @@ public String getHostname() {
     public void setHostname(String server_hostname) {
         hostname = server_hostname;
     }
+
+    public void setApplicationProtocols(String[] protocols) throws IllegalArgumentException {
+        if (protocols == null) {
+            appProtocols = null;
+            return;
+        }
+
+        int index = 0;
+        for (String protocol : protocols) {
+            if (protocol.length() > 255 || protocol.getBytes().length > 255) {
```

Comment:
If the protocol names are ASCII we probably just need to check the `protocol.length()`.